### PR TITLE
Tree node change

### PR DESCRIPTION
**Pull request by fumoffu947**
This is for the convenience of setting children on positions when the list of children is not that long. Like when a binary tree node chould only have a right child but not a left child.

### CODE REVIEW

File: src/tree_node.h [hunk 1]
```diff
@@ -20,6 +20,11 @@ class TreeNode {
 
   void setParent(TreeNode<V>* node);
   void addChild(TreeNode<V>* node);
+  // This method is updating a node and to set for example the right child of
+  // a binary tree without setting the left child.
+  // This method will fill upp the vector with enough cildren if the index is
+  // larger then the size of the vector.
```
Comment (holmgr) [inline src/tree_node.h]: Would personally like a multiline comment instead (/* */) since it is multiple lines.

**Review by holmgr — CHANGES_REQUESTED** [review]
Just a minor nickpick before I approve :)

File: src/tree_node.h [hunk 2]
```diff
@@ -62,6 +69,21 @@ void TreeNode<V>::addChild(TreeNode<V>* node) {
   children.push_back(node);
 }
 
+template<typename V>
+void TreeNode<V>::setChild(TreeNode<V>* node, int index) {
+	if (index < children.size()) {
+		children[index] = node;
+	}
+	else {
+		// I calculate how many empty spaces to add to the list of children.
```
Comment (alexsaaf) [inline src/tree_node.h]: Would like to have "// Calculates ... " instead of "// I calculate" as it describes what the code does, not what you are doing.

**Review by alexsaaf — CHANGES_REQUESTED** [review]
The changes look good, not sure why all the old commits are in the pull request, have they not already been merged? Had one minor request about a comment.

**Comment by fumoffu947** [comment]
All the problems that was identified have been fixed.

**Review by holmgr — APPROVED** [review]
Looks good on my part

**Review by alexsaaf — APPROVED** [review]

**Pull request merged**